modules/juce_audio_devices/native/juce_android_Audio.cpp

```diff
@@ -440,6 +480,46 @@ class AndroidAudioIODevice  : public AudioIODevice,
         }
     }
 
+    void calculateAndReportOutputLatency(int64_t presentationFrame, int64_t presentationTime, int64_t nextOutputFrame, int64_t currentTime)
+    {
+        // Currently, the reported audio output latency on Android is only used for video playback.
+        // Thus, we don't need to report small changes in latency, so we keep the limit at 30 ms.
+        const float latencyChangeLimitSec = 0.03f;
+        const int64_t latencyChangeLimitInSamples = static_cast<int64_t>(outputSampleRate * latencyChangeLimitSec);
+
+        // Calculate when the next frame will be presented.
+        int64_t frameIndexDelta = nextOutputFrame - presentationFrame;
+        int64_t frameTimeDelta = (frameIndexDelta * 1000 * 1000 * 1000) / outputSampleRate;
+        int64_t nextFramePresentationTime = presentationTime + frameTimeDelta;
+        // Latency is the difference between the next frame presentation time and current time.
+        int64_t latencyNs = nextFramePresentationTime - currentTime;
+        int64_t latencySamples = latencyNs * outputSampleRate / (1000 * 1000 * 1000);

```

Comment:
I would do the calculations above using `std::chrono` classes:
```
using namespace std::chrono;
auto frameTimeDelta = duration_cast<nanoseconds>(seconds{ frameIndexDelta }) / outputSampleRate;
auto nextFramePresentationTime = presentationTime + frameTimeDelta;
auto latency = nextPresentationTime  - currentTime;
auto latencySamples = duration_cast<seconds>(latency * outputSampleRate).count();
```

Reply:
Took your suggestion into use almost as-is. Thx!